docker-compose.yml

```diff
@@ -1,47 +1,65 @@
+version: '3.7'
 services:
   web:
     build:
       context: .
+      args:
+        BUILD_ENV: dev
+        POETRY_INSTALL_ARGS: ''
       target: dev
-    init: true
-    # This command just starts the container and keeps it running.
-    # To run a command within the container run docker-compose exec web <command>
-    # A Makefile task exists that starts a development server in the container: make runserver
-    command: tail -f /dev/null
     environment:
-      SECRET_KEY: fake
+      ALLOWED_HOSTS: 127.0.0.1,0.0.0.0,localhost
+      DATABASE_URL: postgres://wagtailio:wagtailio@db:5432/wagtailio # pragma: allowlist secret
       DJANGO_SETTINGS_MODULE: wagtailio.settings.dev
-      DATABASE_URL: postgres://wagtailio:wagtailio@db:5432/wagtailio
-
-      # The env vars get picked up by postgresql client tools (psql, pg_dump, etc)
-      PGHOST: db
-      PGUSER: wagtailio
-      PGPASSWORD: wagtailio
-      PGDATABASE: wagtailio
+      MEDIA_DIR: /app/media/
+      RECAPTCHA_PRIVATE_KEY: UPDATE_THIS # pragma: allowlist secret
+      RECAPTCHA_PUBLIC_KEY: UPDATE_THIS
+      REDIS_URL: redis://redis
+      ENABLE_DJANGO_DEFENDER: 'false'
+      SECRET_KEY: local_dev_secret_key
+      SECURE_SSL_REDIRECT: 'false'
+      STATIC_DIR: /app/static/
+      STATIC_URL: /static/
+    command: tail -f /dev/null # do nothing forever - exec commands elsewhere
     ports:
-      - 8000:8000
+      - 8000:8000 # runserver
     volumes:
-      - .:/app/
+      - ./.git:/app/.git:rw
+      - ./media:/app/media:rw
+      - ./docker:/app/docker:rw
+      - ./wagtailio:/app/wagtailio:rw
+      - ./README.md:/app/README.md:rw
+      - ./fabfile.py:/app/fabfile.py:rw
+
+      # Backend config
+      - ./manage.py:/app/manage.py:rw
+      - ./pyproject.toml:/app/pyproject.toml:rw
+      - ./poetry.lock:/app/poetry.lock:rw
+
     depends_on:
-      db:
-        condition: service_healthy
+      - db
+      - redis
 
   db:
-    image: postgres:13
+    image: postgres:12.3
```

Comment:
nitpick: this sort of blanket change is... not ok.

e.g. staging is on 13.6 so 13 is much better than the downgrade

Reply:
Changed version to 13